x-pack/elastic-agent/pkg/agent/application/gateway/fleet/fleet_gateway.go

```diff
@@ -175,7 +175,8 @@ func (f *fleetGateway) worker() {
 			}
 
 			var errMsg string
-			if err := f.dispatcher.Dispatch(f.acker, actions...); err != nil {
+			// TODO: Is there a purpose in using f.bgContext?
+			if err := f.dispatcher.Dispatch(context.Background(), f.acker, actions...); err != nil {
```

Comment:
context: `fleetGateway` has a `bgContext` struct member

Reply:
I think we could refactor it to remove it and just pass it on call.